frontend/src/container/MetricsApplication/Tabs/util.ts

```diff
@@ -90,12 +90,15 @@ export function onGraphClickHandler(
 	};
 }
 
-export const handleNonInQueryRange = (tags: TagFilterItem[]): TagFilterItem[] =>
+export const handleNonInQueryRange = (
+	tags: TagFilterItem[],
+	isTrace = false,
+): TagFilterItem[] =>
 	tags.map((tag) => {
 		if (tag.op === 'Not IN') {
 			return {
 				...tag,
-				op: 'NIN',
+				op: isTrace ? 'nin' : 'NIN',
```

Comment:
Why is this needed? The `NIN` should work

Reply:
I tried with NIN, it works properly on going to traces from service details page, but was getting this error on subsequently refreshing the traces page in response of query_range.

```
{
  "status": "error",
  "errorType": "bad_data",
  "error": "multiple values [] are not allowed for operator `` for key `signoz.collector.id NIN a1b2c3d4-e5f6-4a5b-9c8d-7e6f5a4b3c2d`"
}
```
Note: signoz.collector.id in the above snippet has been changed to a random UUID.

Reply:
I guess the above issue (i.e. getting error in response of `query_range` on subsequent refresh) is a separate issue and can be addressed separately

Reply:
also, if I manually apply the same filter i.e. signoz.collector.id... using QB, it sets the operator as `nin`